libkineto/src/ActivityProfilerProxy.cpp

```diff
@@ -92,8 +91,7 @@ bool ActivityProfilerProxy::isActive() {
 }
 
 void ActivityProfilerProxy::pushCorrelationId(uint64_t id) {
-  CuptiActivityApi::pushCorrelationID(id,
-    CuptiActivityApi::CorrelationFlowType::Default);
+  controller_->pushCorrelationId(id);
```

Comment:
simple question: why do we want to pass it down into the controller and the activity profilers? I guess you may have a follow-up diff?

Reply:
I share the same question. From your changes, it seems you're aiming to extend the `IActivityProfilerSession` with methods to allow sessions running on `privateuse1` to also access External ID information. However, the current implementation suggests that sessions are intended to serve solely as child profilers. If `privateuse1` employs a custom main profiler, it should be directly registered with the `registerProfiler()` method rather than being added to the session hierarchy using the `configChildProfiler()` method. If this is not the case, then the session should adhere to Kineto's design for profilers, where all manipulations of External IDs should fall under the purview of the main profiler, not child sessions.

This is just a perspective from me outside of Kineto's development, and I hope it doesn't impact your decision-making process if there are any inaccuracies in my understanding. Thank you.

Reply:
> simple question: why do we want to pass it down into the controller and the activity profilers? I guess you may have a follow-up diff?

Some answers in https://github.com/pytorch/kineto/pull/874#discussion_r1508428353. In my plugin, we want to catch events  of EXTERNAL_CORRELATION, so we need to call pushCorrelationId.

Reply:
Actually, I think the new implementation is supposed to have every profiler as a Child Profiler, so your case is okay. In the future we should try to separate out CUPTI and AMD from the main profiler, and into child profiler sessions instead. Thank you for following the recommendations. cc @briancoutinho , @gdankel

Reply:
+1 on this thanks for implementing this as a child module @xunsongh  this could hopefully fix the shortcomings of child profiler architecture today.